Applications/Utils/MeshEdit/ResetPropertiesInPolygonalRegion.cpp

```diff
@@ -192,54 +187,31 @@ int main (int argc, char* argv[])
     GeoLib::Polygon polygon(*(ply));
 
     // *** read mesh
-    MeshLib::Mesh * mesh(MeshLib::IO::readMeshFromFile(mesh_in.getValue()));
-    std::vector<std::string> property_names(
-        mesh->getProperties().getPropertyVectorNames());
-    INFO("Mesh contains %d property vectors:", property_names.size());
-    for (const auto& name : property_names) {
-        INFO("- %s", name.c_str());
-    }
+    auto mesh = std::unique_ptr<MeshLib::Mesh>(
+        MeshLib::IO::readMeshFromFile(mesh_in.getValue()));
     std::string const& property_name(property_name_arg.getValue());
 
     if (char_property_arg.isSet()) {
-        char new_property_val(char_property_arg.getValue());
-
-        // check if PropertyVector exists
-        MeshLib::PropertyVector<char>* pv(nullptr);
-        if (mesh->getProperties().existsPropertyVector<char>(property_name))
-        {
-            pv = mesh->getProperties().getPropertyVector<char>(property_name);
-        }
-        else
-        {
-            pv = mesh->getProperties().createNewPropertyVector<char>(
-                property_name, MeshLib::MeshItemType::Cell, 1);
-            pv->resize(mesh->getElements().size());
-            INFO("Created PropertyVector with name \"%s\".",
-                 property_name.c_str());
-        }
-        resetMeshElementProperty(*mesh, polygon, property_name, new_property_val);
+        resetMeshElementProperty(*mesh, polygon, property_name,
+                                 char_property_arg.getValue());
     }
 
     if (int_property_arg.isSet()) {
-        int int_property_val(int_property_arg.getValue());
-
-        // check if PropertyVector exists
-        auto* pv = mesh->getProperties().getPropertyVector<int>(property_name);
-        if (!pv)
-        {
-            pv = mesh->getProperties().createNewPropertyVector<int>(
-                property_name, MeshLib::MeshItemType::Cell, 1);
-            pv->resize(mesh->getElements().size());
-            INFO("Created PropertyVector with name \"%s\".", property_name.c_str());
-        }
-
-        resetMeshElementProperty(*mesh, polygon, property_name, int_property_val);
+        resetMeshElementProperty(*mesh, polygon, property_name,
+                                 int_property_arg.getValue());
     }
 
     if (bool_property_arg.isSet()) {
         bool bool_property_val(bool_property_arg.getValue());
-        resetMeshElementProperty(*mesh, polygon, property_name, bool_property_val);
+        resetMeshElementProperty(*mesh, polygon, property_name,
+                                 bool_property_arg.getValue());
```

Comment:
Not sure here, but it looks like a bug to me:
I just stumbled on a compiler warning "The `bool_property_val` is not used." Maybe the second parameter in the function call should be `bool_property_val.getValue()` not `..._arg.getValue()`?